src/java_bytecode/java_entry_point.cpp

```diff
@@ -73,7 +74,11 @@ static bool should_init_symbol(const symbolt &sym)
      sym.is_state_var &&
      sym.is_static_lifetime &&
      sym.mode==ID_java)
-    return true;
+  {
+    // Consider some sort of annotation indicating a global variable that
+    // doesn't require initialisation?
+    return !has_suffix(id2string(sym.name), RETURN_VALUE_SUFFIX);
+  }
```

Comment:
Is this necessary? The annotation you could use is marking it `extern`, which will avoid such initialisation in the C front-end. That might work for Java as well, because Java would not be using `is_extern` natively, I believe?

Reply:
Disinclined to abuse the flag so, since I *am* providing a definition of the symbol, it just doesn't need start-of-day initialisation because we know it is always written before reading

Reply:
It just strikes me as something very magical, and very much of unclear value. I accept that it may not be necessary to do the initialisation, but is it really that much harm?

Reply:
At the moment at least with a largeish JAR available you'd get thousands of lines of `__CPROVER_initialize` that you didn't need, and therefore thousands of lines of state in goto-symex. Still, would it make you happier to invent a special `value` (`no_initialisation_required_exprt`?) that makes this clearer?

Reply:
Ok, I get that this may yield many _avoidable_ instructions. What makes me unhappy is this special casing and dependency of otherwise completely unrelated bits of code.

What about renaming the `is_extern` flag and making it no longer C-specific? It could be `do_not_initialize`?

Reply:
Don't want to rename it, but I'll gladly add an extra flag.

Reply:
That would impact (de)serialisation of goto binaries, so take extra care!

Reply:
Argh, then I won't. A comment on the type seems like the least invasive solution in that case.

Reply:
You might just want to tell me to shut up and go with your existing solution plus a TODO saying that a different approach would fall into the class of #1148 :-)

Reply:
Nah that's okay, a type-comment (AKA an annotation, but that's an extra symbol field (which *is* a #1148 matter)) is strictly nicer than what I have now. I'll make a new version of this shortly, as the is-a-stub-callsite thing should be solved better anyhow.